### PR TITLE
Roles support in tests

### PR DESCRIPTION
**Pull request by skonto**
-Added a few role based tests. 
-Added a top project Readme file
-Running with two slaves some checks are not working for example the number of pending tasks in coarse grain mode is not what we expect (i will check if a delay helps but still we have a time out):
TestStarting(Ordinal(0, 2),ClientModeSpec,org.typesafe.spark.mesos.tests.ClientModeSpec,Some(org.typesafe.spark.mesos.tests.ClientModeSpec),simple collect in fine grain mode,simple collect in fine grain mode,Some(MotionToSuppress),Some(LineInFile(25,MesosIntTestHelper.scala)),None,None,main-ScalaTest-running-ClientModeSpec,1447331754928)

TestFailed(Ordinal(0, 3),The test did not complete within the specified 300 seconds time limit.,ClientModeSpec,org.typesafe.spark.mesos.tests.ClientModeSpec,Some(org.typesafe.spark.mesos.tests.ClientModeSpec),simple collect in fine grain mode,simple collect in fine grain mode,Vector(),Some(org.scalatest.exceptions.TestFailedDueToTimeoutException: The test did not complete within the specified 300 seconds time limit.),Some(300165),Some(IndentedText(- simple collect in fine grain mode,simple collect in fine grain mode,0)),Some(SeeStackDepthException),None,None,main-ScalaTest-running-ClientModeSpec,1447332055097)

TestStarting(Ordinal(0, 4),ClientModeSpec,org.typesafe.spark.mesos.tests.ClientModeSpec,Some(org.typesafe.spark.mesos.tests.ClientModeSpec),simple count in coarse grain mode,simple count in coarse grain mode,Some(MotionToSuppress),Some(LineInFile(25,MesosIntTestHelper.scala)),None,None,main-ScalaTest-running-ClientModeSpec,1447332055098)

TestFailed(Ordinal(0, 5),1 did not equal 2 no task should be running,ClientModeSpec,org.typesafe.spark.mesos.tests.ClientModeSpec,Some(org.typesafe.spark.mesos.tests.ClientModeSpec),simple count in coarse grain mode,simple count in coarse grain mode,Vector(),Some(org.scalatest.exceptions.TestFailedException: 1 did not equal 2 no task should be running),Some(36309),Some(IndentedText(- simple count in coarse grain mode,simple count in coarse grain mode,0)),Some(SeeStackDepthException),None,None,main-ScalaTest-running-ClientModeSpec,1447332091407)
-Need to add some testing for attributes

### CODE REVIEW

File: test-runner/src/main/scala/mesostest/mesosstate/StateDump.scala [hunk 1]
```diff
@@ -38,7 +40,43 @@ object MesosCluster {
...
+case class MesosSlave(slaveId:String, resources: Resources, unreservedResources: Resources, usedResources:
+Resources, roleResources:List[ReservedResourcesPerRole])
+
+object MesosSlave {
+
+  def apply(c: Config): MesosSlave = {
+    val slaveId = c.getString("id")
+    import collection.JavaConverters._
+    val reserved= c.getObject("reserved_resources").unwrapped().asScala.
+      map{x=>
```
Comment (dragos) [inline test-runner/src/main/scala/mesostest/mesosstate/StateDump.scala]: nitpick: `map { x =>` (use space between operators)

File: test-runner/src/main/scala/org/typesafe/spark/mesos/tests/ClusterModeSpec.scala [hunk 2]
```diff
@@ -56,6 +56,47 @@ class ClusterModeSpec(mesosConsoleUrl: String)
...
+  runSparkTest("simple count in fine fine grain mode with role in cluster mode",  "spark.mesos.coarse" -> "false",
+    "spark.mesos.role" -> cfg.role, "spark.cores.max" -> cfg.roleCpus) { sc =>
+    val rdd = sc.makeRDD(1 to 5)
+    val res = rdd.sum()
+
+    assert(15 == res)
+
+    val m = MesosCluster.loadStates(mesosConsoleUrl)
+
+    assert(2 == m.frameworks.size, "no task should be running")
```
Comment (skyluc) [inline test-runner/src/main/scala/org/typesafe/spark/mesos/tests/ClusterModeSpec.scala]: wrong message

File: test-runner/src/main/scala/org/typesafe/spark/mesos/tests/ClusterModeSpec.scala [hunk 3]
```diff
@@ -56,6 +56,47 @@ class ClusterModeSpec(mesosConsoleUrl: String)
...
+  runSparkTest("simple count in fine grained mode with role",  "spark.mesos.coarse" -> "true",
+    "spark.mesos.role" -> cfg.role, "spark.cores.max" -> cfg.roleCpus) { sc =>
+    val rdd = sc.makeRDD(1 to 5)
+    val res = rdd.sum()
+
+    assert(15 == res)
+
+    val m = MesosCluster.loadStates(mesosConsoleUrl)
+
+    assert(2 == m.frameworks.size, "no task should be running")
```
Comment (skyluc) [inline test-runner/src/main/scala/org/typesafe/spark/mesos/tests/ClusterModeSpec.scala]: wrong message

File: test-runner/src/main/scala/org/typesafe/spark/mesos/tests/ClientModeSpec.scala [hunk 4]
```diff
@@ -54,5 +56,50 @@ class ClientModeSpec(mesosConsoleUrl: String)
...
+    "spark.mesos.role" -> cfg.role, "spark.cores.max" -> cfg.roleCpus) { sc =>
+    val rdd = sc.makeRDD(1 to 5)
+    val res = rdd.sum()
+
+    assert(15 == res)
+
+    val m = MesosCluster.loadStates(mesosConsoleUrl)
+    assert(m.sparkFramework.isDefined, "only one framework should be running")
+
+    assert(1 == m.sparkFramework.get.tasks.size, "no task should be running")
```
Comment (skyluc) [inline test-runner/src/main/scala/org/typesafe/spark/mesos/tests/ClientModeSpec.scala]: wrong message

**Comment by nraychaudhuri** [comment]
@skonto do you still see test failures? Are you planning to update this PR?

**Comment by skonto** [comment]
i think we should merge it and fix on top of it... it just adds the basic functionality...

**Comment by skyluc** [comment]
LGTM

**Comment by nraychaudhuri** [comment]
@skyluc  is this good to merge?

**Pull request merged**